AsyncDisplayKit/Private/ASDisplayNodeLayout.h

```diff
@@ -0,0 +1,58 @@
+//
+//  ASDisplayNodeLayout.h
+//  AsyncDisplayKit
+//
+//  Created by Michael Schneider on 08/26/16.
+//
+//  Copyright (c) 2014-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under the BSD-style license found in the
+//  LICENSE file in the root directory of this source tree. An additional grant
+//  of patent rights can be found in the PATENTS file in the same directory.
+//
+
+#pragma once
+
+#import "ASDimension.h"
+
+@class ASLayout;
+
+/*
+ * Represents a connection between an ASLayout and a ASDisplayNode
+ * ASDisplayNode uses this to store additional information that are necessary besides the layout
+ */
+struct ASDisplayNodeLayout {
```

Comment:
I think the name represents what it is: A layout that is closely related to `ASDisplayNode`. It’s basically the display node layout. It includes the node as it's primary member and other layout relevant information for this node.
As said, usage is closely related to a `ASDisplayNode` should not be used outside. If we would name it e.g. ASLayoutContext you could also think that you maybe can use it with an `ASLayoutSpec` or other classes that use `ASLayout` somehow.